packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_text_input.tsx

```diff
@@ -0,0 +1,262 @@
+import React from 'react';
+
+import 'components/component_kit/new_designs/cw_text_input.scss';
+import { CWLabel } from '../cw_label';
+import { CWText } from '../cw_text';
+import type { ValidationStatus } from '../cw_validation_text';
+import { getClasses } from '../helpers';
+
+import { ComponentType } from '../types';
+
+type TextInputSize = 'small' | 'large';
+
+export type BaseTextInputProps = {
+  autoComplete?: string;
+  autoFocus?: boolean;
+  containerClassName?: string;
+  defaultValue?: string | number;
+  value?: string | number;
+  iconLeft?: JSX.Element;
+  iconLeftonClick?: () => void;
+  iconRight?: JSX.Element;
+  iconRightonClick?: () => void;
+  inputValidationFn?: (value: string) => [ValidationStatus, string] | [];
+  label?: string | React.ReactNode;
+  maxLength?: number;
+  isCompact?: boolean;
+  name?: string;
+  onInput?: (e) => void;
+  onenterkey?: (e) => void;
+  onClick?: (e) => void;
+  placeholder?: string;
+  tabIndex?: number;
+  manualStatusMessage?: string;
+  manualValidationStatus?: ValidationStatus;
+};
+
+type InputStyleProps = {
+  inputClassName?: string;
+  darkMode?: boolean;
+  disabled?: boolean;
+  size?: TextInputSize;
+  width?: number | string;
+  fullWidth?: boolean;
+  validationStatus?: ValidationStatus;
+  displayOnly?: boolean;
+};
+
+type InputInternalStyleProps = {
+  hasLeftIcon?: boolean;
+  hasRightIcon?: boolean;
+  isTyping?: boolean;
+};
+
+type MessageRowProps = {
+  hasFeedback?: boolean;
+  label: string | React.ReactNode;
+  statusMessage?: string;
+  validationStatus?: ValidationStatus;
+};
+
+type TextInputProps = BaseTextInputProps &
+  InputStyleProps &
+  InputInternalStyleProps &
+  React.HTMLAttributes<HTMLDivElement>;
+
+export const MessageRow = (props: MessageRowProps) => {
+  const { hasFeedback, label, statusMessage, validationStatus } = props;
+
+  return (
+    <div
+      className={getClasses<{ hasFeedback: boolean }>(
+        { hasFeedback },
+        'MessageRow'
+      )}
+    >
+      <CWLabel label={label} />
+      {hasFeedback && (
+        <CWText
+          type="caption"
+          className={getClasses<{ status: ValidationStatus }>(
+            { status: validationStatus },
+            'feedback-message-text'
```

Comment:
I am looking on the figma and validation text is visible under the input. 
![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/e6be71e5-71dd-4794-a766-fa36971f3a3c)

In your solution validation text is above 
![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/322283e9-3422-43dd-9491-548b1577269c)

Also some icons are missing? And the informational text?

Reply:
Addressed validation message, icon and instructional message.